File: app/src/main/java/com/loskon/noteminimalism3/ui/activities/SettingsActivity.java
package com.loskon.noteminimalism3.ui.activities;

import android.app.ListActivity;
import android.content.Intent;
import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

import com.google.android.material.bottomappbar.BottomAppBar;
import com.loskon.noteminimalism3.R;
import com.loskon.noteminimalism3.auxiliary.other.AppFontManager;
import com.loskon.noteminimalism3.auxiliary.other.MyColor;
import com.loskon.noteminimalism3.auxiliary.sharedpref.GetSharedPref;
import com.loskon.noteminimalism3.ui.fragments.SettingsFragment;

/**
 * Класс для работы с основными настройками
 */

public class SettingsActivity extends AppCompatActivity {

    private BottomAppBar btmAppBarSettings;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        MyColor.setDarkTheme(GetSharedPref.isDarkMode(this));
        AppFontManager.setFont(this);
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_settings);
        MyColor.setColorStatBarAndTaskDesc(this);

        if (savedInstanceState == null) {
            getSupportFragmentManager()
                    .beginTransaction()
                    .replace(R.id.fragment_settings, new SettingsFragment())
                    .commit();
        }

        btmAppBarSettings = findViewById(R.id.btmAppBarSettings);
        btmAppBarSettings.setNavigationOnClickListener(v -> onBackPressed());


    }

    public void fff() {
        Intent intent = new Intent(this, ListActivity.class);
        intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK | Intent.FLAG_ACTIVITY_NEW_TASK);
        startActivity(intent);
    }

    @Override
    protected void onResume() {
        super.onResume();
        MyColor.setNavIconColor(this, btmAppBarSettings);
    }
}

File: app/src/main/java/com/loskon/noteminimalism3/ui/activities/SettingsActivity.kt
package com.loskon.noteminimalism3.ui.activities

import android.os.Bundle
import android.view.Menu
import androidx.coordinatorlayout.widget.CoordinatorLayout
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import com.google.android.material.bottomappbar.BottomAppBar
import com.loskon.noteminimalism3.R
import com.loskon.noteminimalism3.managers.FontManager
import com.loskon.noteminimalism3.managers.setMenuIconsColor
import com.loskon.noteminimalism3.managers.setNavigationIconColor
import com.loskon.noteminimalism3.sharedpref.AppPreference
import com.loskon.noteminimalism3.ui.fragments.RootSettingsFragment
import com.loskon.noteminimalism3.ui.snackbars.WarningSnackbar

/**
 * Хост представления для фрагментов
 */

class SettingsActivity : BaseActivity() {

    private lateinit var coordLayout: CoordinatorLayout
    private lateinit var bottomBar: BottomAppBar
    private lateinit var fragmentManager: FragmentManager
    private lateinit var menu: Menu

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_settings)
        setupViewDeclaration()
        initObjects()
        establishViewsColor()
        openSettingsFragment(savedInstanceState)
        setupViewsListeners()
        configureBottomBarMenu()
    }

    private fun setupViewDeclaration() {
        coordLayout = findViewById(R.id.coord_layout_settings)
        bottomBar = findViewById(R.id.bottom_bar_settings)
    }

    private fun initObjects() {
        fragmentManager = supportFragmentManager
        menu = bottomBar.menu
    }

    private fun establishViewsColor() {
        val color: Int = AppPreference.getColor(this)
        establishViewsColor(color)
    }

    private fun establishViewsColor(color: Int) {
        bottomBar.setNavigationIconColor(color)
        menu.setMenuIconsColor(color)
    }

    private fun openSettingsFragment(savedInstanceState: Bundle?) {
        if (savedInstanceState == null) {
            fragmentManager
                .beginTransaction()
                .add(R.id.fragment_container_settings, RootSettingsFragment())
                .commit()
        }
    }

    private fun setupViewsListeners() {
        bottomBar.setNavigationOnClickListener {
            onBackPressed()
        }
    }

    private fun configureBottomBarMenu() {
        changeVisibilityMenuItem(false)
    }

    //----------------------------------------------------------------------------------------------
    fun replaceFragment(fragment: Fragment) {
        fragmentManager
            .beginTransaction()
            .replace(R.id.fragment_container_settings, fragment)
            .addToBackStack(null)
            .commit()
    }

    fun showSnackbar(messageType: String) {
        WarningSnackbar.show(coordLayout, bottomBar, messageType)
    }

    fun onChangeColor(color: Int) {
        establishViewsColor(color)
    }

    fun changeVisibilityMenuItem(isVisible: Boolean) {
        menu.findItem(R.id.action_account).isVisible = isVisible
    }

    fun setAppFonts() {
        FontManager.setFont(this)
    }

    //----------------------------------------------------------------------------------------------
    fun getBottomBar(): BottomAppBar {
        return bottomBar
    }
}